### ISSUE TITLE
[BUG] Cannot create Image from URL with query string

### ISSUE THREAD

**Issue by tserong** [harvester/harvester#5186]
**Describe the bug**
When creating a VM image, the Harvester UI tries to automatically check the file extension to make sure it's supported. It looks like it's just taking everything after the last `.` in the URL to check the extension. This breaks when you need to download from a URL which has a query string, for example:

![image](https://github.com/harvester/harvester/assets/754700/0bb2c44b-62e5-4fb8-ab4b-35c3339033f2)

**To Reproduce**
1. Go to 'Image: Create'
2. Enter the URL of an ISO or qcow2 image, which has a query string after the extension, e.g.: https://updates.suse.com/SUSE/Products/SLE-Product-Minimal/15-SP5-QU2/x86_64/SLES15-SP5-Minimal-VM.x86_64-Cloud-QU2.qcow2?AAAAAAAAAAAAAA
3. Click "Create"
4. Image creation will fail with the error "The URL you have entered ends in an extension that we do not support. We only accept image files that end in .img, .iso, .qcow, .qcow2, .raw.

**Expected behavior**
Image creation succeeds.

**Support bundle**
N/A

**Environment**
 - Harvester ISO version: approximately v1.3
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Virtualized

**Additional context**
The validator code is at https://github.com/harvester/dashboard/blob/9c96e1f47105081ea740d741fca8ca6cba462869/pkg/harvester/validators/vm-image.js#L5-L22, and all it's doing is taking everything after the last `/` character as the file name, then everything after the last `.` in that as the extension. It might be enough - in the case where the URL contains a `?` character - to simply strip the `?` and everything after it before doing the thing with the slash and the dot.

Or... Do we really need to validate that the URL contains an extension at all? Sure, most URLs of images you're going to download will have extensions, but that's not necessarily guaranteed (there's no requirement that the path part of a URL actually includes an extension). If the URL doesn't contain a recognizable extension, we already don't automatically fill in the Name field, and the user has to enter a name manually. Why not just take the extension from the Name field, rather than trying to get it from the URL?

**Comment by Vicente-Cheng** [comment]
still alive

**Comment by tserong** [comment]
Unless I'm mistaken, this should actually now be fixed in v1.7.0

**Comment by innobead** [comment]
Yes, this seems to be missed from the sprint process of 1.7.0.. cc @Vicente-Cheng 

Let's make sure this goes through the 1.7.1 QA process.

**Comment by innobead** [comment]
@a110605, can you help @devsymo move this issue to 'Ready for Testing' so QA can work on it in the next sprint? Thanks.

cc @harvester/tpm

**Comment by Vicente-Cheng** [comment]
Since, the fix was merged moved to `Ready for Testing`

**Comment by asc5543** [comment]
## Test result
Test PASS, the image can be created without issue.

## Environment
* Harvester ISO version: `v1.7.1-dev-20260104`
* Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): `2 node via ipxe-example`


## Test Scenario
### Test Steps
1. <details>
   <summary> ✔ Go to 'Image: Create'
   </summary>

    <img width="2546" height="1300" alt="Image" src="https://github.com/user-attachments/assets/75c8ca37-9054-4b39-ae36-c5a24ef21346" />
    <img width="2546" height="1300" alt="Image" src="https://github.com/user-attachments/assets/77f4c083-6971-43cf-9e26-184004a83ddf" />

   </details>

1. <details>
   <summary> ✔ Enter the URL of an ISO or qcow2 image, which has a query string after the extension, then click "Create"
   </summary>
  
    <img width="2541" height="1321" alt="Image" src="https://github.com/user-attachments/assets/bddd2ce6-26a9-4564-9a14-158b15adb160" />
    <img width="2541" height="1321" alt="Image" src="https://github.com/user-attachments/assets/1f720d81-738c-4006-a080-44baf3cffaae" />
    
   </details>

1. <details>
   <summary> ✔ The image is created w/o any issue
   </summary>

    <img width="2541" height="1321" alt="Image" src="https://github.com/user-attachments/assets/517d810c-a9a4-4eb9-9c80-1f2d53d4a27e" />
    <img width="2541" height="1321" alt="Image" src="https://github.com/user-attachments/assets/b4f692d0-0f95-4880-b80b-ace4d5fec4d4" />

   </details>
